contrib/parseq-examples/src/main/java/com/linkedin/parseq/example/batching/BatchingClientExample.java

```diff
@@ -0,0 +1,70 @@
+/* $Id$ */
+package com.linkedin.parseq.example.batching;
+
+import java.util.List;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.batching.BatchingSupport;
+import com.linkedin.parseq.example.common.AbstractExample;
+import com.linkedin.parseq.example.common.ExampleUtil;
+import com.linkedin.parseq.promise.Promises;
+
+
+/**
+ * @author Jaroslaw Odzga (jodzga@linkedin.com)
+ */
+public class BatchingClientExample extends AbstractExample {
+
+  public static class ExampleBatchingStrategy extends BatchingStrategy<Long, Long> {
+
+    @Override
+    public Long groupForKey(Long key) {
```

Comment:
I'm not convinced this is the best way of doing grouping. It seems to me like you'd have two very common cases:

1) All the keys can be batched together. For instance, in the case of fetching profiles we would always have the profile id. We would write the batch get method to take a list of profile ids and fetch a list of profiles. In this case it's just extra work to override the `groupForKey` method.

2) The keys have some additional parameter added to them. For instance, you want to get a profile given a profile id and a viewer id. You can only batch get profile ids together if they have the same viewer id. Therefore, your key would be a tuple (viewer id, profile id). Given a list of tuples like:
(A, 1)
(A, 2)
(B, 1)
you'd group them into:
(A, List(1,2))
(B, List(1))
In this case you'd want the group type G to be a single object and the taskForBatch method to take G and a List of profile ids, NOT a list of tuples (because all the first elements in the tuple would be the same).